src/lib/pickles_types/pcs_batch.ml

```diff
@@ -89,12 +89,18 @@ let combine_split_commitments _t ~scale_and_add ~init:i ~xi (type n)
         ~f:(fun { With_degree_bound.unshifted; shifted } ->
           Array.to_list unshifted @ [ shifted ] )
   in
-  match List.rev flat with
-  | [] ->
-      failwith "combine_split_commitments: empty"
-  | init :: comms ->
-      List.fold_left comms ~init:(i init) ~f:(fun acc p ->
-          scale_and_add ~acc ~xi p )
+  let rec go = function
+    | [] ->
+        failwith "combine_split_commitments: empty"
+    | init :: comms -> (
+        match i init with
+        | None ->
+            go comms
+        | Some init ->
+            List.fold_left comms ~init ~f:(fun acc p ->
```

Comment:
This new function could behave differently than the previous one. Do we have a rationale or a test that shows why this is so?